### PR TITLE
Add resource labels to metrics

### LINKED ISSUE #478: Ability to read labels from pod/deployements being scanned and add to prometheus metrics
In our clusters we have specific labels that we'd like to add to the metrics generated by `trivy-operator`.

Namely we'd like the ability to assign owner label to these metrics since trying to join these metrics with other metrics to get the owner is non-trivial.

IE, trivy scans doom pod/deployment, then reads owner label. Adds that to the metrics generated.

Let me know if you need more clarification here.

### PR DESCRIPTION
**Pull request by hlastras**
## Description
Add, by configuration, scanned resources labels into the prometheus metrics with a configurable prefix.

## Related issues
- Close #478

## Checklist
- [x] I've read the [guidelines for contributing](https://github.com/aquasecurity/trivy-operator/blob/main/CONTRIBUTING.md) to this repository.
- [x] I've added tests that prove my fix is effective or that my feature works.
- [x] I've updated the [documentation](https://github.com/aquasecurity/trivy-operator/tree/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

**Comment by hlastras** [comment]
The biggest changes have been in the `collector.go` file.

I had to do a little refactor to make it workable. I have moved all the prometheus descriptors, labels and severities that were in global variables, to a struct that will go inside the `ResourcesMetricsCollector`.
This is because the labels are no longer fixed, but instead vary according to the configuration, so they must be generated once we have the configuration data.
I have added a `constructor` for `ResourcesMetricsCollector` to build the labels and the descriptors based on the configuration.

I don't know if I have added enough tests, or if I should add any integration tests.
I haven't added any documentation yet until I'm sure the implementation will look like this.

I have tried to make small meaningful commits instead of one big one, to be able to see the evolution of the code. (even knowing that in the end there will be squashed).
First time using Conventional Commit.

And last but not least, yes, the names I have chosen are probably not the best, so I am totally open to changing them if there are suggestions.

**Comment by hlastras** [comment]
Hi @chen-keinan thanks for taking the time to review it.

Yes, I could move the configuration into the configmap if that fits better, the only thing is that I have to add that config into the [metrics collector](https://github.com/aquasecurity/trivy-operator/blob/main/pkg/metrics/collector.go#L186) because currently it only has the env var configurations (etc.Config).  
If you agree with that, I'll move it
`report.resourceLabels` and `report.resourceLabelsPrefix` fit for you?

**Comment by chen-keinan** [comment]
> Hi @chen-keinan thanks for taking the time to review it.
> 
> Yes, I could move the configuration into the configmap if that fits better, the only thing is that I have to add that config into the [metrics collector](https://github.com/aquasecurity/trivy-operator/blob/main/pkg/metrics/collector.go#L186) because currently it only has the env var configurations (etc.Config). If you agree with that, I'll move it `report.resourceLabels` and `report.resourceLabelsPrefix` fit for you?

yes, please  go for it

**Comment by hlastras** [comment]
I think it's perfect, I'm going with it

**Comment by hlastras** [comment]
Hi @chen-keinan 
I have already made all the changes you suggested, although I'm not very confident that the description of the configurations is very good, so any suggestion is welcome.

On the other hand, I've never done anything with Helm before, so I have no idea if my changes to the template are correct.

Thanks in advance

File: pkg/metrics/collector.go [hunk 1]
```diff
@@ -318,39 +402,45 @@ func (c *ResourcesMetricsCollector) collectRbacAssessmentReports(ctx context.Con
 		for _, r := range reports.Items {
 			labelValues[0] = r.Namespace
 			labelValues[1] = r.Name
-			c.populateRbacAssessmentValues(labelValues, rbacAssessmentDesc, r.Report.Summary, metrics, 2)
+			for i, label := range c.GetReportResourceLabels() {
+				labelValues[i+3] = r.Labels[label]
```
Comment (chen-keinan) [inline pkg/metrics/collector.go:406]: I see that in resources that do not have the label , an empty value in shown in metrics , wdyt ? 
`trivy_role_rbacassessments{k8s_label_owner="",name="role-868458b9d6",namespace="kube-system",severity="Critical"} 1`

**Comment by chen-keinan** [comment]
@hlastras helm values look good , I have added to comments

File: pkg/vulnerabilityreport/controller.go [hunk 2]
```diff
@@ -305,6 +307,7 @@ func (r *WorkloadController) submitScanJob(ctx context.Context, owner client.Obj
 		WithPodSecurityContext(scanJobSecurityContext).
 		WithSecurityContext(scanJobContainerSecurityContext).
 		WithPodTemplateLabels(scanJobPodTemplateLabels).
+		WithResourceLabelsToInclude(resourceLabelsToInclude).
```
[the remark this answers is not in the document]
Comment (hlastras) [reply pkg/vulnerabilityreport/controller.go]: After reviewing it carefully, I agree that it is not necessary.

For some reason I thought that when building the reports we only had the information of the Job, that's why I added it there, but indeed in the builder we have the controller that is the object that has been analyzed and from which I can get (and in fact I get) the labels values. I guess when I started I thought that this object was the job and that's why I copied the labels to the job.
My fault.

I have already made a commit to remove that unnecessary part.

**Comment by chen-keinan** [comment]
LGTM 🚀

**Review by chen-keinan — APPROVED** [review]

**Pull request merged**